lib/hooks.tsx

```diff
@@ -50,14 +67,20 @@ export function useConversation<T extends Talk.UIBox>(
   box: T | undefined,
   syncConversation: ConversationProps["syncConversation"],
   conversationId: ConversationProps["conversationId"],
+  asGuest: boolean | undefined,
 ) {
   const conversation = useMemo(() => {
     if (typeof syncConversation === "function") {
       return session?.isAlive ? syncConversation(session) : undefined;
     }
     return syncConversation ?? conversationId;
   }, [session, syncConversation, conversationId]);
-  useSetter(box, conversation, "select");
+
+  const args = (
+    conversation !== undefined ? [conversation, { asGuest }] : []
```

Comment:
shouldn't this be

```suggestion
    conversation !== undefined ? [conversation, { asGuest }] : [conversation]
```

?

Reply:
Looking at it again I think you're right that it's a bug. However, I don't think it should be `[]` nor `[conversation]`, but rather `undefined`.

We used to have `useSetter` instead of `useSpreadSetter`. The JSDoc comment on `useSetter` says (bolded by me):

> Calls method `setter` on `box` with `value`, **iff `value` is set** & deeply different from before (and the box is alive)

`useSetter` implementation:

```tsx
value = usePreviousIfDeeplyEqual(value);
useEffect(() => {
  if (value !== undefined && box?.isAlive) {
    box[setter](value); // In useSpreadSetter it's: box[setter](...value);
  }
}, [setter, box, value]);
```

So if `conversation` was `undefined`, `select` wouldn't get called at all. I wanted to preserve this behavior with `useSpreadSetter` which is why I (mistakenly) used `[]`.